packages/data/addon/mirage/handlers/graphql.js

```diff
@@ -19,41 +80,173 @@ function _getSeedForRequest(table, args, fields) {
+ *
+ * @param {Object[]} filters - time dimension filter objects with operator, values, and grain
+ * @returns {Array<Moment|null>} - lowerbound and upperbound of moments for interval
+ */
+function _createInterval(filters) {
+  const intervals = filters.map(f => DATE_FILTER_OPS[f.operator](f.values, f.grain));
+  const lowerLimits = intervals.map(interval => interval[0]).filter(Boolean);
+  const lowerBound = lowerLimits.length ? moment.max(lowerLimits) : null;
+
+  const upperLimits = intervals.map(interval => interval[1]).filter(Boolean);
+  const upperBound = upperLimits.length ? moment.min(upperLimits) : null;
+
+  if (!upperBound && !lowerBound) {
+    return [];
+  }
+
+  return !lowerBound || !upperBound || lowerBound.isSameOrBefore(upperBound) ? [lowerBound, upperBound] : [];
+}
+
+/**
+ * @param {Object[]} filters - time dimension filter objects with operator, values, and grain
+ * @returns {Object} - date intervals for each time dimension
  */
-function _getDates(/* filter */) {
-  // TODO: Generate dates based on filters on time dimensions and the chosen grain
-  let day = moment();
-  let days = [];
-  for (let i = 0; i < 3; i++) {
-    days.push(moment(day).format(API_DATE_FORMAT));
-    day = moment(day).subtract(1, 'days');
+function _intervalsForFilters(filters) {
+  // Group filters by their field without grain and assign grain and filterWithoutGrain properties
+  const filtersWithGrain = filters.reduce((byField, filter) => {
+    const [, fieldWithoutGrain, grain] = TIME_DIMENSION_REGEX.exec(filter.field);
+    const filterObject = {
+      ...filter,
+      fieldWithoutGrain,
+      grain
+    };
+    byField[fieldWithoutGrain] = [...(byField[fieldWithoutGrain] || []), filterObject];
+    return byField;
+  }, {});
+
+  return Object.keys(filtersWithGrain).reduce((acc, curr) => {
+    acc[curr] = _createInterval(filtersWithGrain[curr]);
+    return acc;
+  }, {});
+}
+
+/**
+ * @param {Array<Moment|null>} interval
+ * @param {String} grain
+ * @returns {Moment[]} - all the dates for the interval in the supplied time grain
+ */
+function _datesForInterval(interval, grain) {
+  if (!interval[0] && !interval[1]) {
+    return [];
+  }
+  let start;
+  let end;
+  const dates = [];
+
+  // Default interval to at most 1 month long if start or end are null
+  if (!interval[0]) {
+    end = moment(interval[1]).startOf(grain);
+    start = moment(end).subtract(1, 'month'); //Default start to 1 month before end
+  } else if (!interval[1]) {
+    start = interval[0].startOf(grain);
+    let monthAhead = moment(start).add(1, 'month');
+    let current = moment();
+    end = monthAhead.isSameOrBefore(current) ? monthAhead.startOf(grain) : current.startOf(grain); //Default end to current or 1 month past start, whichever is closer to start
+  } else {
+    start = interval[0].startOf(grain);
+    end = interval[1].startOf(grain);
+  }
+  for (let i = start; i.isSameOrBefore(end); i.add(1, grain)) {
```

Comment:
I think we can remove this and the `startOf(grain)` logic and rely on [getDatesForInterval](https://github.com/yahoo/navi/blob/893642dac365059d1d5cf374b8784431cc687173/packages/core/addon/utils/date.js#L27) One other thing that takes into account is the difference between `week`/`isoweek`. I'm not sure which elide is reporting on but with fili it's the isoweek

Reply:
Yeah unfortunately navi-data doesn't have access to the interval util class since navi-core depends on navi-data. Good call on isoWeek though. I'm sure elide uses that as well.

Reply:
Oops, yea I thought we used that for the bard mock